geonode_mapstore_client/client/js/components/home/FeaturedList.jsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { Spinner, Button } from 'react-bootstrap-v1';
+import HTML from '@mapstore/framework/components/I18N/HTML';
+import ResourceCard from '@js/components/home/ResourceCard';
+import FaIcon from '@js/components/home/FaIcon';
+import { withResizeDetector } from 'react-resize-detector';
+
+const Cards = withResizeDetector(({
+    resources,
+    formatHref,
+    isCardActive,
+    width,
+    buildHrefByTemplate,
+    options,
+    size,
+    margin,
+    count
+}) => {
+    const cardWidth = width >= size + margin * 2
+        ? Math.floor((width - margin * count) / count)
+        : '100%';
+
+    const ulPadding = Math.floor(margin / 2);
+    const isSingleCard = count === 0 || count === 1;
+
+    const gridLayoutSpace = (idx) => {
+
+        const gridSpace = isSingleCard
+            ? {
+                width: width - margin,
+                margin: ulPadding
+            }
+            : {
+                width: cardWidth,
+                marginRight: (idx + 1) % count === 0 ? 0 : margin,
+                marginTop: margin
+            };
+
+        return gridSpace;
+    };
+
+    const containerStyle = isSingleCard
+        ? {
+            paddingBottom: margin
+        }
+        : {
+            paddingLeft: ulPadding,
+            paddingBottom: margin
+        };
+    return (
+        <ul
+            style={containerStyle}
+        >
+            {resources.map((resource, idx) => {
+                return (
+                    <li
+                        key={resource?.pk}
+                        style={(gridLayoutSpace(idx))}
+                    >
+                        <ResourceCard
+                            active={isCardActive(resource)}
+                            data={resource}
+                            formatHref={formatHref}
+                            options={options}
+                            buildHrefByTemplate={buildHrefByTemplate}
+                            layoutCardsStyle="grid"
+                        />
+                    </li>
+                );
+            })}
+        </ul>
+    );
+});
+
+const FeaturedList = withResizeDetector(({
+    resources,
+    loading,
+    isNextPageAvailable,
+    formatHref,
+    isCardActive,
+    containerStyle,
+    header,
+    cardOptions,
+    pageSize,
+    buildHrefByTemplate,
+    isPreviousPageAvailable,
+    loadFeaturedResources,
+    detectedWidth,
+    containerWidth
+}) => {
+
+    const width = containerWidth || detectedWidth;
+    const margin = 24;
+    const size = 320;
+    const count = Math.floor(width / (size + margin));
```

Comment:
In this case we are wrapping the component with withResizeDetector but not using width provided by this HOC. There is something that should be checked here. The width we need is this one:

![image](https://user-images.githubusercontent.com/19175505/121508463-eadd5680-c9e5-11eb-8bcb-c520181d9645.png)

and withResizeDetector should compute that value so please try to remove detectedWidth and containerWidth props and use width prop. We should also move the initial load of resources here and if the count changes we should trigger a new request 

```js
    width,
    onLoad
}) => {

    const margin = 24;
    const size = 320;
    const count = Math.floor(width / (size + margin));
    useEffect(() => {
        onLoad(undefined, count);
    }, [count]);
```